Test books

Drocheg opened the pull request:
-Adds and configure respect
-Adds Faker
-Adds FactoryBot
-Adds DatabaseCleaner
-Adds tests for book
-Ignores rubocop block length in test files

Trello: https://trello.com/c/4puP9ceQ/23-crear-tests-para-el-modelo-de-book

Drocheg committed:
Adds rspect and create tests for book

-Adds and configure respect
-Adds Faker
-Adds FactoryBot
-Adds DatabaseCleaner
-Adds tests for book

Drocheg committed:
Fix all rubocop violations

Drocheg committed:
Fix others rubcopo violation

-Ignores block length in test files
-Changes configuration nesting in rails_helper.rb

Gmarn commented on spec/models/book_spec.rb:
```diff
@@ -0,0 +1,80 @@
+
+describe Book do
+  it do
+    should validate_presence_of(:genre)
```
At the end you let it here? Better if you not let useless and duplicate tests. So better if you remove it.

Gmarn replied:
All shoulld ;)

Gmarn commented on spec/models/book_spec.rb:
```diff
@@ -0,0 +1,80 @@
+      end
+    end
+
+    context 'When the genre is nil' do
```
To have a DRYier code you can try to write a test with an `each`:
```ruby
%i(genre author ...).each do |field|
  # Your context and test
end
```

Gmarn commented on spec/models/book_spec.rb:
```diff
@@ -0,0 +1,80 @@
+    end
+
+    context 'When the genre is nil' do
+      it do
```
Do not forget to add description on you `it` when you can:
`it 'description' do`

Gmarn commented on spec/rails_helper.rb:
```diff
@@ -0,0 +1,78 @@
+  # config.filter_gems_from_backtrace("gem name")
+
+  # enable render views
+  config.render_views
```
You can comment this because you are doing an API without views.

Gmarn commented on spec/factories.rb:
```diff
@@ -0,0 +1,12 @@
+
+FactoryBot.define do
+  factory :book do
+    genre Faker::HitchhikersGuideToTheGalaxy.location
```
For factories with Faker we used to use block, this allow factory to not have static value during all the testing process.
To able this behaviour define your faker in `{ }`
like:
```ruby
genre     { Faker::HitchhikersGuideToTheGalaxy.location }
author    { Faker::Name.name }
publisher { Faker::Name.name }
# ...
```
And even better if you align each block as above.

Drocheg committed:
Multiple changes to have better tests

-Deletes duplicated tests
-Changes so test are created in a .each to make code DRY
-Doesn't render view anymore
-Adds Description to "it"s
-Factory initializations changes to blocks

Gmarn commented on spec/models/book_spec.rb:
```diff
@@ -3,77 +3,20 @@
-        book = build(:book, year: nil)
-        expect(book).to be_invalid
+    %i[genre author image title publisher year].each do |field|
+      context 'When the ' + field.to_s + 'is nil' do
```
To concatenate string is better behaviour to use:
`" smth #{code} smth"`

Drocheg committed:
Changes how a string is concatenated

Drocheg committed:
Automatic fix of rubocop offence
